Feature: auto-scroll, auto-flip

Guzmazow opened the pull request:
Closes #10 and all duplicates

Everything is off by default and new UI is hidden

New buttons above the current navigation buttons:
<img width="400" height="366" alt="image" src="https://github.com/user-attachments/assets/891e5cc1-e335-4c9d-b02c-406ac8477b58" />

New configuration subsection:
<img width="425" height="884" alt="image" src="https://github.com/user-attachments/assets/078fe174-e553-4df7-94be-f773980c6bfe" />

Guzmazow committed:
Add ability to auto-scroll and auto-flip pages on button click

Guzmazow committed:
Fix auto-scroll increments to prettier numbers
Remove useless logging

BrutuZ commented:
Could the scroll work and be labeled on comparable units, like `1 screen (height) every X seconds`?
Pixels per whatever a screen refresh might be a little too obtuse for the average user.

Guzmazow commented:
i can try, but i cannot guarantee that it won't affect fluidity...

AntsyLich commented:
I can understand the resume icon but what is the + and - icon supposed to do?

Guzmazow committed:
Make auto-scroll speeds human readable

Guzmazow committed:
Make refreshRate retrieval future-proof

Guzmazow committed:
spotless code cleanup

MajorTanya commented:
> Strange locally the migration tests are fine <img alt="image" width="1311" height="443" src="https://private-user-images.githubusercontent.com/12956451/468339539-1f74a90c-e74f-453b-9812-8a864d727e72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI5ODUwNDMsIm5iZiI6MTc1Mjk4NDc0MywicGF0aCI6Ii8xMjk1NjQ1MS80NjgzMzk1MzktMWY3NGE5MGMtZTc0Zi00NTNiLTk4MTItOGE4NjRkNzI3ZTcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIwVDA0MTIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkZDQ5YzAxZDk4Njg1NGQ3MTkyNWZhZTA0MzFmZGYyZWYyNzM5NmVmZDg5ZTI5YjRiM2UzNzVhMzYxMmY0MzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gjq_6W-hz9lvdewOjq1TSe5m6uxEVSH0MoCA2y0D7QU">

that one is known to be flaky for some godsforsaken reasons

BrutuZ commented:
Question out of ignorant curiosity: VRR won't be a problem if the scroll is based on refresh rate?

Guzmazow commented:
> Question out of ignorant curiosity: VRR won't be a problem if the scroll is based on refresh rate?

yes if the change of refresh rate happens during auto-scroll. low battery -> lower refresh rate -> lower scroll speed (2x in my case)
the speed is recalculated every start of scrolling so it is enough to just start scrolling again

Guzmazow committed:
auto-scroll VariableRefreshRate support

Guzmazow commented:
added VRR support duing scrolling

Guzmazow committed:
do not launch automation thread if automation is disabled
refactor config to correct subtype

Guzmazow committed:
added initial values to config, fix automation start

MajorTanya commented:
I see you used the Android Log thing. The project uses logcat overall, see many other places that have logcat calls across the project. :+1:

Not sure if Antsy will merge this but I must say I like the smallness of the diff for this feature :+1:

AntsyLich commented:
> Not sure if Antsy will merge this [...]

I would've closed it instead of making comments

MajorTanya commented:
> > Not sure if Antsy will merge this [...]
> 
> I would've closed it instead of making comments

Somehow missed your earlier comments 😅

Guzmazow committed:
Option to limit automation duration. Stop automation on last chapter end, sleep, backgrounding. Code refactor.

Guzmazow commented:
added option to specify maximum duration/chapters
fixed background/sleep glitch
refactored so the automation code is minimal in the huge viewer files

AntsyLich commented:
Make the current value string smaller in slider settings

Guzmazow committed:
simplify automation preference summary text

Guzmazow committed:
moved localization with numbers to plurals

Guzmazow committed:
moved localization with numbers to plurals

Guzmazow commented:
<img width="425" height="884" alt="image" src="https://github.com/user-attachments/assets/24028d52-0e17-4377-9fb1-2128aa45d9f5" />

Guzmazow commented:
on the way i made the translation pluralized
<img width="420" height="592" alt="image" src="https://github.com/user-attachments/assets/3da6a8a5-fd43-486a-b87d-578d28c7a369" />

Guzmazow committed:
Keep screen on during automation

AntsyLich commented:
I'll do a full review after the next stable release

Smol-Ame commented:
@ElainaaTech if you're asking about how to use this feature in the app, you can't. This pull request is still pending review & approval, and whenever the dev releases a new version of the app with said feature in the new version.

If you don't have anything useful to comment about the pull request (feedback, changes, questions), wait patiently.

Guzmazow commented:
rebased on main/head

Guzmazow commented:
> A great option to have. One question that came to my mind - how would the behaviour be for the double-page/wide pages?

No idea.... but theroretically now it works as if you flip or scroll by hand
